google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -1756,31 +1772,90 @@ def default_acl
         #
         # @return [Policy] the current Cloud IAM Policy for this bucket
         #
-        # @example
+        # @example Retrieving a Policy that is implicitly version 1:
         #   require "google/cloud/storage"
         #
         #   storage = Google::Cloud::Storage.new
-        #
         #   bucket = storage.bucket "my-todo-app"
         #
         #   policy = bucket.policy
+        #   policy.version # 1
+        #   puts policy.roles["roles/storage.objectViewer"]
         #
-        # @example Retrieve the latest policy and update it in a block:
+        # @example Retrieving a version 3 Policy using `requested_policy_version`:
         #   require "google/cloud/storage"
         #
         #   storage = Google::Cloud::Storage.new
+        #   bucket = storage.bucket "my-todo-app"
+        #
+        #   policy = bucket.policy requested_policy_version: 3
+        #   policy.version # 3
+        #   puts policy.bindings.find do |b|
+        #     b[:role] == "roles/storage.objectViewer"
+        #   end
         #
+        # @example Updating a Policy that is implicitly version 1:
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
         #   bucket = storage.bucket "my-todo-app"
         #
         #   bucket.policy do |p|
-        #     p.add "roles/owner", "user:owner@example.com"
+        #     p.version # 1
+        #     p.remove "roles/storage.admin", "user:owner@example.com"
+        #     p.add "roles/storage.admin", "user:newowner@example.com"
+        #     p.roles["roles/storage.objectViewer"] = ["allUsers"]
+        #   end
+        #
+        # @example Updating a Policy from version 1 to version 3 by adding a condition:
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #   bucket = storage.bucket "my-todo-app"
+        #
+        #   bucket.policy requested_policy_version: 3 do |p|
+        #     p.version # 1
+        #     p.version = 3 # Must be explicitly set to opt-in to support for conditions.
+        #     p.bindings.insert({
+        #                         role: "roles/storage.admin",
+        #                         members: ["user:owner@example.com"],
+        #                         condition: {
+        #                           title: "test-condition",
+        #                           description: "description of condition",
+        #                           expression: "expr1"
```

Comment:
One last note, it would useful to show an example expression here:
```
resource.name.startsWith("projects/_/buckets/bucket-name/objects/prefix-a-")
```

Reply:
Escaping the double quotes is OK, correct?

```rb
expression: "resource.name.startsWith(\"projects/_/buckets/bucket-name/objects/prefix-a-\")"
```

Reply:
Full example. Let me know if this looks OK and I'll make the change.

```rb
bucket.policy requested_policy_version: 3 do |p|
  p.version # 1
  p.version = 3 # Must be explicitly set to opt-in to support for conditions.

  expr = "resource.name.startsWith(\"projects/_/buckets/bucket-name/objects/prefix-a-\")"
  p.bindings.insert({
                      role: "roles/storage.admin",
                      members: ["user:owner@example.com"],
                      condition: {
                        title: "test-condition",
                        description: "description of condition",
                        expression: expr
                      }
                    })
end
```

Reply:
Yep you can pass in escaped double quotes! This LGTM.